### DISCUSSION TITLE
[Discussions] What is the practical sense for `needRotate` boolean?

### DISCUSSION THREAD

**Discussion by bambinoua** [CaiJingLong/dart_image_size_getter#32] [Q&A] [answered]
### Content

What is the practical sense for `needRotate` boolean? If I just need to know the size of the image? Width and height are enough to know image orientation - portrait or landscape.

**Comment by CaiJingLong** [comment, marked as answer]
The exif of jpeg maybe contains orientation attr.

A example to know it: https://developer.mozilla.org/en-US/docs/Web/CSS/image-orientation

See the attr info to read [wiki](https://en.wikipedia.org/wiki/Exif).

<img width="1050" alt="image" src="https://user-images.githubusercontent.com/14145407/223643761-accad08c-1292-4b70-af2c-a11f211ad6af.png">

This value range is 1~8, but only 1, 3, 6 and 8 can be seen when using.

A document of MIT: https://www.media.mit.edu/pia/Research/deepview/exif.html#orientation

<img width="1073" alt="image" src="https://user-images.githubusercontent.com/14145407/223646676-c853c636-b98c-4c4d-b7bb-ddd1c2f67830.png">

From load to render graph:

```mermaid
graph TD;
    A[Load Image] --> B{exif orientation = 6 or 8?};
    B -- 6 --> C[Rotate counterclockwise by 90 degrees];
    B -- 8 --> D[Rotate clockwise by 90 degrees];
    B -- other values --> E[No rotation];
    C --> F[Render rotated image];
    D --> F;
    E --> F;
```

This is image size getter about the attribute:

```mermaid
graph TD;
    A[Read metadata] --> B[Get image width and height];
    B --> C[Determine if rotation is needed];
    C --> D{Need rotation?};
    D -- Yes --> E[Rotate image];
    D -- No --> F[Do not rotate image];
    E --> G[ Return Size with needRotation=true];
    F --> H[Return Size with needRotation=false];

```
